felix/rules/static.go

```diff
@@ -1065,6 +1065,16 @@ func (r *DefaultRuleRenderer) StaticManglePostroutingChain(ipVersion uint8) *gen
 	// mangle table is typically used, if at all, for packet manipulations that might need to
 	// apply to our allowed traffic.
 
+	ipConf := r.ipSetConfig(ipVersion)
+	allIPsSetName := ipConf.NameForMainIPSet(IPSetIDAllPools)
+	rules = append(rules, generictables.Rule{
+		Match: r.NewMatch().
+			SourceIPSet(allIPsSetName).
+			NotDestIPSet(allIPsSetName),
+		Action:  r.Jump(ChainQoSPolicy),
+		Comment: []string{"QoS policy for traffic leaving cluster"},
+	})
```

Comment:
This needs to be enabled only for iptables/nftables but not for ebpf (most likely depending on the solution used in ebpf dataplane)